scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,286 @@
+        return sms
+    elseif summoner:getWeather() == dsp.weather.AURORAS and element == dsp.subEffect.LIGHT_DAMAGE then
+        return ms
+    elseif summoner:getWeather() == dsp.weather.STELLAR_GLARE and element == dsp.subEffect.LIGHT_DAMAGE then
+        return sms
+    elseif summoner:getWeather() == dsp.weather.GLOOM and element == dsp.subEffect.DARKNESS_DAMAGE then
+        return ms
+    elseif summoner:getWeather() == dsp.weather.DARKNESS and element == dsp.subEffect.DARKNESS_DAMAGE then
+        return sms
+    elseif summoner:getWeather() == dsp.weather.SNOW and element == dsp.subEffect.WIND_DAMAGE then
+        return ms * - 1
+    elseif summoner:getWeather() == dsp.weather.BLIZZARDS and element == dsp.subEffect.WIND_DAMAGE then
+        return sms * - 1
+    elseif summoner:getWeather() == dsp.weather.WIND and element == dsp.subEffect.EARTH_DAMAGE then
+        return ms * - 1
+    elseif summoner:getWeather() == dsp.weather.GALES and element == dsp.subEffect.EARTH_DAMAGE then
+        return sms * - 1
+    elseif summoner:getWeather() == dsp.weather.DUST_STORM and element == dsp.subEffect.LIGHTNING_DAMAGE then
+        return ms * - 1
+    elseif summoner:getWeather() == dsp.weather.SAND_STORM and element == dsp.subEffect.LIGHTNING_DAMAGE then
+        return sms * - 1
+    elseif summoner:getWeather() == dsp.weather.THUNDER and element == dsp.subEffect.WATER_DAMAGE then
+        return ms * - 1
+    elseif summoner:getWeather() == dsp.weather.THUNDERSTORMS and element == dsp.subEffect.WATER_DAMAGE then
+        return sms * - 1
+    elseif summoner:getWeather() == dsp.weather.RAIN and element == dsp.subEffect.FIRE_DAMAGE then
+        return ms * - 1
+    elseif summoner:getWeather() == dsp.weather.SQUALL and element == dsp.subEffect.FIRE_DAMAGE then
+        return sms * - 1
+    elseif summoner:getWeather() == dsp.weather.HOT_SPELL and element == dsp.subEffect.ICE_DAMAGE then
+        return ms * - 1
+    elseif summoner:getWeather() == dsp.weather.HEAT_WAVE and element == dsp.subEffect.ICE_DAMAGE then
+        return sms * - 1
+    elseif summoner:getWeather() == dsp.weather.AURORAS and element == dsp.subEffect.DARKNESS_DAMAGE then
+        return ms * - 1
+    elseif summoner:getWeather() == dsp.weather.STELLAR_GLARE and element == dsp.subEffect.DARKNESS_DAMAGE then
+        return sms * - 1
+    elseif summoner:getWeather() == dsp.weather.GLOOM and element == dsp.subEffect.LIGHT_DAMAGE then
+        return ms * - 1
+    elseif summoner:getWeather() == dsp.weather.DARKNESS and element == dsp.subEffect.LIGHT_DAMAGE then
+        return sms * - 1
+    end
+    return 0
+end;
+
+function getDayMod(avatar)
+    local element = getElement(avatar)
+    local ms = -3000    
+    local summoner = avatar:getMaster()    
+    if VanadielDayElement() == dsp.day.ICEDAY and element == dsp.subEffect.ICE_DAMAGE then
```

Comment:
tables my dude. tables.
```lua
local myTableNameHere =
{
    [index here] = what value we want from selecting it,
    [index here] = what value we want from selecting it,
    [index here] = what value we want from selecting it
}

if myTable[input to get index here] then
```

Reply:
pretty much anytime you have to type "else" that many times, you are working to hard

Reply:
nvm, didn't realize this was your other PR re sent in, I expected the new commits to replace what was on the other one (github will auto update the pr if the branch's content changes, so force pushing to overwrite it works)